src/hotspot/share/opto/vectornode.hpp

```diff
@@ -904,8 +904,7 @@ class StoreVectorMaskedNode : public StoreVectorNode {
  public:
   StoreVectorMaskedNode(Node* c, Node* mem, Node* dst, Node* src, const TypePtr* at, Node* mask)
    : StoreVectorNode(c, mem, dst, at, src) {
-    assert(mask->bottom_type()->isa_vectmask(), "sanity");
```

Comment:
Why the assert was added before? And why you can remove it now?

Reply:
Does this mean the `mask` input can be a normal `vect_type ` like the mask input of `VectorBlend` for `LoadVectorMasked/StoreVectorMasked` over X86 AVX2 systems? They do not depend on the predicated feature for the AVX2 systems, right?

Reply:
IR was specifically added for predicated targets (AVX512 and ARM's SVE),  this patch is re-using this IR for non-predicated AVX2 target where mask could be a vector type.

Reply:
Got it.